iree/tools/opt_main.cc

```diff
@@ -0,0 +1,149 @@
+//===- mlir-opt.cpp - MLIR Optimizer Driver -------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Main entry function for mlir-opt for when built as standalone binary.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/InitLLVM.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/ToolOutputFile.h"
+#include "mlir/Analysis/Passes.h"
+#include "mlir/InitAllDialects.h"
+#include "mlir/InitAllPasses.h"
```

Comment:
We should use our own curated list instead of still having these? Do you want to do that in a follow-up?

Reply:
I think that's the plan. Let's keep this PR small

Reply:
Yeah, that would be a thing for a follow up. Still figuring out how `tf_mlir_opt_main` does this (it still links in `@llvm-project//mlir:AllPassesAndDialects`) and how to keep this consistent with CMake.